pkg/images/backend/knhelpers.go

```diff
@@ -0,0 +1,192 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package backend
+
+import (
+	"encoding/json"
+
+	"github.com/go-openapi/strfmt"
+	knbuild "github.com/knative/build/pkg/apis/build/v1alpha1"
+	"github.com/pkg/errors"
+	dapi "github.com/vmware/dispatch/pkg/api/v1"
+	"github.com/vmware/dispatch/pkg/utils"
+	"github.com/vmware/dispatch/pkg/utils/knaming"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	defaultDockerfile         = "Dockerfile"
+	defaultSystemPackagesFile = "system-packages.txt"
+	defaultPackagesFile       = "packages.txt"
+)
+
+// FromBaseImage produced Knative BuildTemplate from Dispatch BaseImage
+func FromBaseImage(baseimage *dapi.BaseImage) *knbuild.BuildTemplate {
+	if baseimage == nil {
+		return nil
+	}
+
+	// TODO: BaseImage -> BuildTemplate make this right and configurable
+	return &knbuild.BuildTemplate{
```

Comment:
I am not sure if I fully understand the idea behind backing a `BaseImage` with a `BuildTemplate`. Seems like we could get away with a single instance `BuildTemplate` for a given version of dispatch.

Reply:
In the legacy dispatch, `BaseImage` was an allowed golden image created by an admin and all images were its derivatives. If we just need that functionality, the set of allowed BaseImages could very well be defined in a config map.

Reply:
Thanks! So, a better solution might be to treat Dispatch base-image as a parameter, which is a URL to the admin image. And store all allowed base image in a config map. The Knative Build (Dispatch Image) will only reference to the base_image URL. And we only need one BuildTemplate, Dispatch Image mapping to Knative Build CRD remains the same. Does this make sense?